src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaXmlWriter.cs

```diff
@@ -280,6 +742,27 @@ internal override void WriteBinaryTypeAttributes(IEdmBinaryTypeReference referen
             }
         }
 
+        /// <summary>
+        /// Asynchronously writes BinaryType attributes.
+        /// </summary>
+        /// <param name="reference">The Edm binary type reference.</param>
+        /// <returns>Task represents an asynchronous operation.</returns>
+        internal override Task WriteBinaryTypeAttributesAsync(IEdmBinaryTypeReference reference)
+        {
+            if (reference.IsUnbounded)
+            {
+                return this.WriteRequiredAttributeAsync(CsdlConstants.Attribute_MaxLength, CsdlConstants.Value_Max, EdmValueWriter.StringAsXml);
+            }
+            else

```

Comment:
The `else` is unnecessary since it would not be reached if the if conditioned passed.

Reply:
I have removed the unnecessary `else`